### ISSUE TITLE
Suggestion about the mouse behaviors

### ISSUE THREAD

**Issue by trmdi** [kupiqu/plasma-active-application#7]
It should behave like Latte's panel: double click to maximize/restore, support drag
Currently, a click will minimize the window, it's very easy to make users trigger that action accidentally.

@psifidotos could help you in this issue.

**Comment by kupiqu** [comment]
Yes. I have actually talked to @psifidotos. Goods news for users is that he has a very nice roadmap for improving this applet, including this very mouse behavior, and UI and more (https://www.reddit.com/r/kde/comments/9uh5yc/window_buttons_applet_v01/e96ma92). Soon I will transfer the ownership of the applet to him to boost development. In the meantime, I will try to focus on issue #5.

**Comment by kupiqu** [comment]
> It should behave like Latte's panel: double click to maximize/restore

Done in master (commit 8ec650b66c9d0f4211e66a2259cac3454645959f; keeping middle click to close, and no minimize option anymore). Dragging will have to wait... Leaving this issue open.

**Comment by trmdi** [comment]
If you want, you could make an option like "Mouse handle by this applet"
- Default: on -> use your own implement. This is for the case users use this applet on a Plasma panel.
- When the option is off -> set `main.enabled = true` -> Latte panel handles the mouse events

There could be a better option. What do you think about this @psifidotos ?

**Comment by psifidotos** [comment]
> What do you think about this @psifidotos ?

ok...  Let me explain you the architecture and how I think it must be implemented...

Latte in git version provides an Communicator Engine in order to communicate with applets:
the files are:
Communicator Engine->Latte Bridge->Actions

the Communicator is the main class and can be found at: https://phabricator.kde.org/source/latte-dock/browse/master/containment/package/contents/ui/applet/communicator/Engine.qml

the Latte Bridge is the object that is passed as it in any applet that has set
`property QtObject latteBridge: null`
at its root item. the latte bridge is located at: https://phabricator.kde.org/source/latte-dock/browse/master/containment/package/contents/ui/applet/communicator/LatteBridge.qml

and finally Actions is just a way to send events from any applet to LatteBridge and then at Engine etc. etc....

Actions is located at: https://phabricator.kde.org/source/latte-dock/browse/master/containment/package/contents/ui/applet/communicator/Actions.qml

-----

The plan is to add in the Actions item some functions for dragging....
An example about how latteBridge can be used for another use case is the Window Buttons applet... check out the main.qml file and search for Latte in order to see the magic behind it...

**Comment by kupiqu** [comment]
mmm, @psifidotos, if you don't mind, I think I'll leave this to you ;-)

**Comment by psifidotos** [comment]
> mmm, @psifidotos, if you don't mind, I think I'll leave this to you ;-)

yeah dont worry...
